lib/omniauth/strategies/shopify.rb

```diff
@@ -49,7 +49,7 @@ def valid_signature?
       def valid_scope?(token)
         params = options.authorize_params.merge(options_for("authorize"))
         return false unless token && params[:scope] && token['scope']
-        (params[:scope].split(',').sort == token['scope'].split(',').sort)
+        (params[:scope].delete(' ').split(',').sort == token['scope'].delete(' ').split(',').sort)
```

Comment:
So it is a problem with `params[:scope]` in that case the workaround would be to remove spaces from scope used to build the middleware, and we can provide a proper test for this.